server/src/main/java/org/cloudfoundry/identity/uaa/scim/endpoints/ScimUserEndpoints.java

```diff
@@ -224,7 +250,34 @@ public ScimUser createUser(@RequestBody ScimUser user, HttpServletRequest reques
             passwordValidator.validate(user.getPassword());
         }
 
-        ScimUser scimUser = scimUserProvisioning.createUser(user, user.getPassword(), identityZoneManager.getCurrentIdentityZoneId());
+        user.setZoneId(identityZoneManager.getCurrentIdentityZoneId());
+
+        if (!aliasHandler.aliasPropertiesAreValid(user, null)) {
+            throw new ScimException("Alias ID and/or alias ZID are invalid.", HttpStatus.BAD_REQUEST);
+        }
+
+        // create the user and an alias for it if necessary
+        ScimUser scimUser;
+        try {
+            scimUser = transactionTemplate.execute(txStatus -> {
```

Comment:
you know I dont like transactions in UAA, but if you really want use them, can you please use them in scenario where this feature is enabled ( aliasEntitiesEnabled == true ) and all other UAA users (aliasEntitiesEnabled == false) create a user with one call only.

Reply:
fixed with commit ced0186

Reply:
ok